dht/token.hh

```diff
@@ -218,6 +218,16 @@ inline bool operator<=(const token& t1, const token& t2) { return std::rel_ops::
 inline bool operator>=(const token& t1, const token& t2) { return std::rel_ops::operator>=(t1, t2); }
 std::ostream& operator<<(std::ostream& out, const token& t);
 
+// Returns a successor for token t.
+// The caller must ensure there is a next token, otherwise
+// the result is unspecified.
+//
+// Precondition: t.kind() == dht::token::kind::key
+inline
+token next_token(const token& t) {
+    return {dht::token::kind::key, t._data + 1};
```

Comment:
This is specified as a precondition:

```
The caller must ensure there is a next token, otherwise
the result is unspecified.
```

If _data is max int, there is no next token, so precondition is not satisfied.